### DISCUSSION TITLE
using vault in semaphore

### DISCUSSION THREAD

**Discussion by rogertheshruber** [semaphoreui/semaphore#1505] [Q&A]
i can see on the task template that it asks for vault password.  i'm not sure how to use that password to get values out of the vault or how to use those values in my playbook.  also is it possible to pass credentials from the keystore securly into playbooks?  i need to have a username and password combination to pass to vmware or other virtualization platforms.

**Comment by ansibleguy** [comment]
Greetings.

You might want to look into Ansible-Vault usage: [Ansible docs](https://docs.ansible.com/ansible/latest/vault_guide/index.html)

There you can store secrets encrypted and they get decrypted by Ansible on runtime

**Comment by rogertheshruber** [comment]
i know how to use the vault from command line.  could not figure it out with semaphore.  i ended up adding to the CLI Args box in the template like below.

[
  "-e",
  "@/var/tmp/testing1/testvault.yml"
]

**Comment by alessandro-celli** [comment]
I have the same problem, i don't understand the right string associated with del value. I try with the name of my secret in semaphore but doesn't work.

**Comment by lwandrebeck** [comment]
I use key store to store vault passwords, using login with password type, and empty login. In the task template, use this key as vault password. HTH.
